src/lpython/parser/tokenizer.re

```diff
@@ -72,28 +71,22 @@ void lex_int(Allocator &al, const unsigned char *s,
 {
     if (std::tolower(s[1]) == 'x') {
         s = s + 2;
+        lex_dec_int_large(al,
+            (const unsigned char *) get_value(al, (char *)s, 16), e, u);
```

Comment:
There is a bug in here,
it gives `0` as output or malloc failed:
Example 1:
```python
123
0x1ABC
```
```console
$ lpython --show-ast --new-parser examples/expr2.py
runtime_error: malloc failed.
```
Example 2:
```python
123
-123
-0x123
0x1ABC
```
```console
$ lpython --show-ast --new-parser --indent examples/expr2.py
(Module [
   (Expr 
      (ConstantInt 123 ())) 
   (Expr 
      (UnaryOp 
         USub 
         (ConstantInt 123 ()))) 
   (Expr 
      (UnaryOp 
         USub 
         (ConstantInt 0 ())))                            <----------
   (Expr 
      (ConstantInt 0 ()))] [])
```